src/main/scala/mesosphere/marathon/SchedulerDriverFactory.scala

```diff
@@ -27,10 +33,34 @@ class MesosSchedulerDriverFactory @Inject() (
     * As a side effect, the corresponding driver is set in the [[MarathonSchedulerDriverHolder]].
     */
   override def createDriver(): SchedulerDriver = {
-    implicit val zkTimeout = config.zkTimeoutDuration
-    val frameworkId = Await.result(frameworkIdRepository.get(), zkTimeout).map(_.toProto)
+    val frameworkId: Option[FrameworkID] = MesosSchedulerDriverFactory.getFrameworkId(
+      crashStrategy, config.zkTimeoutDuration, frameworkIdRepository, instanceRepository)
     val driver = MarathonSchedulerDriver.newDriver(config, httpConfig, scheduler, frameworkId)
     holder.driver = Some(driver)
     driver
   }
 }
+
+object MesosSchedulerDriverFactory extends StrictLogging {
+  def getFrameworkId(
+    crashStrategy: CrashStrategy,
+    zkTimeout: FiniteDuration,
+    frameworkIdRepository: FrameworkIdRepository,
+    instanceRepository: InstanceRepository)(implicit mat: Materializer): Option[FrameworkID] = {
+
+    def instancesAreDefined: Boolean = Await.result(instanceRepository.ids().runWith(Sink.headOption), zkTimeout).nonEmpty
+
+    Await.result(frameworkIdRepository.get(), zkTimeout).map(_.toProto) match {
+      case frameworkId @ Some(_) =>
+        frameworkId
+      case None if instancesAreDefined =>
+        logger.error("Refusing to create a new Framework ID while there are existing instances.\n" +
+          "Please see for an explanation of the issue, and how to recover: https://mesosphere.github.io/marathon/docs/framework-id.html")
+        Await.result(crashStrategy.crash(CrashStrategy.FrameworkIdMissing), Duration.Inf)
+        throw new RuntimeException("Refusing to allow creation of a new Framework ID")
```

Comment:
Is this exception handled if we crash before?

Reply:
No; we throw an exception to prevent the creation fo a framework.